how-it-works/1_protocol/02_consensus.subpage.md

```diff
@@ -5,14 +5,118 @@ shareImage: /img/how-it-works/consensus.600.jpg
+A *block maker*, i.e., a node that is supposed to create the block for the current round, is determined using a cryptographic mechanism called *random beacon* explained below.
+The so-determined block maker assembles a block from (a subset of) the ingress messages submitted directly to the node or received from other nodes through broadcast and xnet messages sent to this subnet from other subnets.
+Xnet messages are pulled from the respective sending subnets by the block maker.
+A block, once assembled, is then broadcast to all nodes of the subnet using P2P as a block proposal for the current round.
+
+If the network is unreliable or the chosen block maker corrupted or other things are going wrong with producing, broadcasting, and notarizing the block, the first proposed block may not get notarized within a reasonable timeframe or not at all.
+Thus, other nodes also need to get the opportunity to act as block makers and propose a block, so that eventually a block for the round will be successfully proposed and notarized, even under adversarial conditions, as more and more nodes attempt to propose a block.
+
+The block makers for a round are chosen through a random permutation of the nodes of the subnet based on randomness derived from the *random beacon*.
+The random beacon is a random value computed as a unique BLS threshold signature over the random beacon value of the previous round:
+A qualifying set of nodes compute and broadcast (using P2P) a BLS threshold signature share on the previous random beacon and, irrespective of which subset of shares are combined, the combined shares result in the same threshold signature value as long as a qualifying quorum has provided their shares.
+A random permutation is computed using the random beacon as source of randomness
+The permutation imposes an unpredictable and unbiasable ranking of block makers for the round, ranging from the rank-$0$ to the rank-$(n-1)$ block maker, where $n$ is the number of nodes in the subnet.
+The highest priority (rank-$0$) block maker is the first one to propose a block in this round.
+If this block does not get fully notarized within a given time period, the next-lower-priority (rank-$1$) block maker takes an attempt at producing a block.
+This process continues with increasing-rank block makers until all nodes for the round are exhausted to attempt to produce a block.
+Eventually, the subnet will be able to generate a fully-notarized block if every node has had the opportunity to propose one.
+In the scenario where the subnet works without problems, the rank-$0$ block maker's block is the only block proposed and getting notarized in the round.
+This is the best case and leads to the lowest latency and highest througput in the subnet.
+
+## Notarization
+
+If a node receives a block proposal broadcast by a block maker for the round, it validates the block for syntactic correctness and validity of the signatures of the messages contained in the block.
```

Comment:
Are there signatures on messages contained in the block. Maybe I missed it, but where are such signatures discussed?